sourcegraph/sourcegraph
# bext: update private repo sync banners

Closes [issues/31920](https://github.com/sourcegraph/sourcegraph/issues/31920).

See also PR https://github.com/sourcegraph/sourcegraph/pull/31690 discussion.

Show different instructions on the options page for repo not found error depending on Sourcegraph URL and whether the user is admin.

| Sourcegraph instance  | User is admin | Screenshot | Notes |
| ------------------------------ | ------------------ | ---------------- | -------- |
| Cloud                           |                        |  ![Screenshot from 2022-03-24 15-38-02](https://user-images.githubusercontent.com/25318659/160400906-9be3f796-2332-448a-acdc-16accb7488f3.png)   |  Sourceraph button text is 'Configure on Sourcegraph'. It navigates to user settings manage repositories page with current repo name as filter query.         |
| Self-hosted / managed |  ✅                 |  ![Screenshot from 2022-03-24 14-23-46](https://user-images.githubusercontent.com/25318659/160401284-37a6f3f5-66e5-4590-af22-622f706e8a9e.png)  | - Link in the extension popup navigates user to [add repos docs page](https://docs.sourcegraph.com/admin/repo/add)<br> - Sourceraph button text is 'Repository not found'. It still [links to the repository](https://github.com/sourcegraph/sourcegraph/blob/main/client/browser/src/shared/code-hosts/shared/ViewOnSourcegraphButton.tsx#L134), but will show 'Not Found' page. This information is just for context, the button is not the matter of change of this PR. |
| Self-hosted / managed |  ❌                 | ![Screenshot from 2022-03-24 15-37-27](https://user-images.githubusercontent.com/25318659/160401219-590f00d7-7e36-4900-93ba-e1b0e4979153.png)  | ⬆️ See notes on the Sourcegraph button for the on-prem / managed instance admin user. | 
 
UPD: updated copies according to [this comment](https://github.com/sourcegraph/sourcegraph/pull/31922#issuecomment-1081577004)
![repo sync](https://user-images.githubusercontent.com/25318659/160581974-08ab9cd2-16e7-4681-bd4c-626954fb36a8.png)


## Test plan
 - `sg run bext`
 - Sourcegraph cloud instance
   - set Sourcegraph URL to Sourcegrpah Cloud
   - log in to your cloud account
   - go to the private repository page which is not synced in user settings 
   - check the private repo message on the options page (see screenshots above)
 - Sourcegraph on-prem / managed instance
   - set Sourcegrpah URL to on-prem / managed instance
   - log in as non-admin user
   - check not-synced repo message on the options page (see screenshots above)
   - log in as admin user
   - check not-synced repo message for the admin user (see screenshots above)


## App preview:
- [Link](https://sg-web-taras-yemets-bext-update-private.onrender.com)
